frontend/catalyst/pennylane_extensions.py

```diff
@@ -191,12 +191,22 @@ def _make_jaxpr_check_differentiable(f: Differentiable, grad_params: GradParams,
                     return_ops.append(eq.primitive)
                     break
 
-        if method == "ps" and any(prim not in [expval_p, probs_p] for prim in return_ops):
+        assert isinstance(
+            f, qml.QNode
+        ), "Differentiation methods other than finite-differences can only operate on a QNode"
+        if f.diff_method is None:
+            raise ValueError(
+                "Cannot differentiate a QNode explicitly marked non-differentiable (with"
+                " diff_method=None)"
+            )
+
+        has_invalid_return = any(prim not in [expval_p, probs_p] for prim in return_ops)
+        if f.diff_method == "parameter-shift" and has_invalid_return:
             raise TypeError(
                 "The parameter-shift method can only be used for QNodes "
                 "which return either qml.expval or qml.probs."
             )
-        if method == "adj" and any(prim not in [expval_p] for prim in return_ops):
+        if f.diff_method == "adjoint" and has_invalid_return:
```

Comment:
Note that parameter-shift and adjoint enforce different restrictions on the circuit.